test/test.erl

```diff
@@ -24,7 +24,7 @@ known_problem_should_pass_test_() ->
                 V -> V
             catch E -> nok
             end,
-        ?_assertEqual(nok, Result)
+        ?_assertMatch({nok, _}, {Result, File})
```

Comment:
Is this for improving the error reporting if the assertion fails?

Reply:
yes, exactly. this is a bit of poor-man's, badmatch solution, but good enough for now.

example before
```
**error:{assertEqual,[{module,test},
              {line,27},
              {expression,"Result"},
              {expected,nok},
              {value,ok}]}
```
after
```
**error:{assertMatch,[{module,test},
              {line,27},
              {expression,"{ Result , File }"},
              {pattern,"{ nok , _ }"},
              {value,{ok,"test/known_problems/should_pass/map_union.erl"}}]}
```